cmd/finch/version.go

```diff
@@ -4,27 +4,117 @@
 
+	"github.com/runfinch/finch/pkg/command"
+	"github.com/runfinch/finch/pkg/flog"
+	"github.com/runfinch/finch/pkg/lima"
 	"github.com/runfinch/finch/pkg/version"
 )
 
-var finchVersion = version.Version
-
-func newVersionCommand() *cobra.Command {
+func newVersionCommand(limaCmdCreator command.LimaCmdCreator, logger flog.Logger, stdOut io.Writer) *cobra.Command {
 	versionCommand := &cobra.Command{
 		Use:   "version",
 		Args:  cobra.NoArgs,
 		Short: "Shows Finch version information",
-		RunE:  versionAction,
+		RunE:  newVersionAction(limaCmdCreator, logger, stdOut).runAdapter,
 	}
 
 	return versionCommand
 }
 
-func versionAction(cmd *cobra.Command, args []string) error {
-	fmt.Println("Finch version:", finchVersion)
+type versionAction struct {
+	creator command.LimaCmdCreator
+	logger  flog.Logger
+	stdOut  io.Writer
+}
+
+// NerdctlVersionOutput captures the nerdctl version.
+type NerdctlVersionOutput struct {
+	Client NerdctlClientOuput  `json:"Client"`
+	Server NerdctlServerOutput `json:"Server"`
+}
+
+// NerdctlClientOuput captures the nerdctl Client output.
+type NerdctlClientOuput struct {
+	Version    string                    `json:"Version"`
+	GitCommit  string                    `json:"GitCommit"`
+	GoVersion  string                    `json:"GoVersion"`
+	Os         string                    `json:"Os"`
+	Arch       string                    `json:"Arch"`
+	Components []NerdctlComponentsOutput `json:"Components"`
+}
+
+// NerdctlServerOutput captures the nerdctl Server output.
+type NerdctlServerOutput struct {
+	Components []NerdctlComponentsOutput `json:"Components"`
+}
+
+// NerdctlComponentsOutput captures the nerdctl components output.
+type NerdctlComponentsOutput struct {
+	Name    string `json:"Name"`
+	Version string `json:"Version"`
+	Details struct {
+		GitCommit string `json:"GitCommit"`
+	}
+}
+
+func newVersionAction(creator command.LimaCmdCreator, logger flog.Logger, stdOut io.Writer) *versionAction {
+	return &versionAction{creator: creator, logger: logger, stdOut: stdOut}
+}
+
+func (va *versionAction) runAdapter(cmd *cobra.Command, args []string) error {
+	return va.run()
+}
+
+func (va *versionAction) run() error {
+	status, err := lima.GetVMStatus(va.creator, va.logger, limaInstanceName)
+	//nolint: gocritic // will change to switch statement later
+	if err != nil {
+		fmt.Fprintf(va.stdOut, "Finch version:\t%s", version.Version)
+		return err
+	} else if status != lima.Running {
+		fmt.Fprintf(va.stdOut, "Finch version:\t%s", version.Version)
+	} else {
+		limaArgs := []string{"shell", limaInstanceName, "nerdctl", "version", "--format", "json"}
+		out, err := va.creator.CreateWithoutStdio(limaArgs...).Output()
+		if err != nil {
+			fmt.Fprintf(va.stdOut, "Finch version:\t%s", version.Version)
+			return err
+		}
+
+		var nerdctlVersion NerdctlVersionOutput
+		err = json.Unmarshal(out, &nerdctlVersion)
+
+		if err != nil {
+			fmt.Fprintf(va.stdOut, "Finch version:\t%s", version.Version)
+			return err
```

Comment:
Could we prefix the errors?

```suggestion
			return fmt.Errorf("failed to create the nerdctl version command: %w", err)
		}

		var nerdctlVersion NerdctlVersionOutput
		err = json.Unmarshal(out, &nerdctlVersion)
		if err != nil {
			fmt.Fprintf(va.stdOut, "Finch version:\t%s", version.Version)
			return fmt.Errorf("failed to JSON-unmarshal the nerdctl version output: %w", err)
```